src/api/Migrations/20181204172354_FasterLocationQuery.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace SearchAndCompare.Migrations
+{
+    public partial class FasterLocationQuery : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql(@"DROP FUNCTION course_distance(double precision,
+                                                                 double precision,
+                                                                 double precision);");
+
+            migrationBuilder.Sql(@"CREATE OR REPLACE FUNCTION course_distance( lat DOUBLE PRECISION, 
+                                                            lon DOUBLE PRECISION, 
+                                                            rad DOUBLE PRECISION) 
+                                            RETURNS TABLE (""Id"" integer,
+                                                           ""Distance"" double precision,
+                                                           ""DistanceAddress"" text) AS $$
+                    WITH distances as (
+                        SELECT ""Id"", ""Address"",
+                            111045.0* DEGREES(ACOS(COS(RADIANS(lat))
+                                    * COS(RADIANS(""Latitude""))
+                                    * COS(RADIANS(lon) - RADIANS(""Longitude""))
+                                    + SIN(RADIANS(lat))
+                                    * SIN(RADIANS(""Latitude"")))) AS ""Distance""
+                        from location
+                        where 111045.0* DEGREES(ACOS(COS(RADIANS(lat))
+                                    * COS(RADIANS(""Latitude""))
+                                    * COS(RADIANS(lon) - RADIANS(""Longitude""))
+                                    + SIN(RADIANS(lat))
+                                    * SIN(RADIANS(""Latitude"")))) <  rad
+                    ), matching_campuses AS (
+                        select campus.""CourseId"" as ""Id"", ""Distance"", ""Address"" as ""DistanceAddress""
+                        from campus
+                        JOIN distances on distances.""Id"" = ""LocationId""    
+                    ), matching_courses AS (
+                        select course.""Id"", ""Distance"", ""Address"" as ""DistanceAddress""
+                        from course
+                        JOIN distances on distances.""Id"" = ""ProviderLocationId""
+                    ), matching_all AS (
+                        SELECT ""Id"", ""Distance"", ""DistanceAddress"" from matching_campuses
+                        
+                        UNION ALL
+
+                        SELECT ""Id"", ""Distance"", ""DistanceAddress"" from matching_courses
+
+                    ), matching_all_with_rows AS (
+                        SELECT ""Id"", ""Distance"", ""DistanceAddress"", ROW_NUMBER() OVER(PARTITION BY ""Id"" ORDER BY ""Distance"" ASC) as ordinal
+                        FROM matching_all
+                    )
```

Comment:
love the use of CTEs, makes the progressive build of logic much clearer than the original